nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -649,6 +740,60 @@ async fn ssc_start_running_snapshot(
     Ok(map)
 }
 
+async fn ssc_start_running_snapshot_undo(
+    sagactx: NexusActionContext,
+) -> Result<(), anyhow::Error> {
+    let log = sagactx.user_data().log();
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
+
+    let snapshot_id = sagactx.lookup::<Uuid>("snapshot_id")?;
+    info!(log, "Undoing snapshot start running request for {snapshot_id}");
+
+    // Lookup the regions used by the source disk...
+    let (.., disk) = LookupPath::new(&opctx, &osagactx.datastore())
+        .disk_id(params.disk_id)
+        .fetch()
+        .await?;
+    let datasets_and_regions =
+        osagactx.datastore().get_allocated_regions(disk.volume_id).await?;
+
+    // ... and instruct each of those regions to delete the running snapshot.
+    for (dataset, region) in datasets_and_regions {
+        let url = format!("http://{}", dataset.address());
+        let client = CrucibleAgentClient::new(&url);
+
+        use crucible_agent_client::Error::ErrorResponse;
+        use http::status::StatusCode;
+
+        client
+            .region_delete_running_snapshot(
+                &RegionId(region.id().to_string()),
+                &snapshot_id.to_string(),
+            )
+            .await
+            .map(|_| ())
+            // NOTE: If we later create a volume record and delete it, the
+            // running snapshot may be deleted (see:
+            // ssc_create_volume_record_undo).
+            //
+            // To cope, we treat "running snapshot not found" as "Ok", since it
+            // may just be the result of the volume deletion steps completing.
+            .or_else(|err| match err {
+                ErrorResponse(r) if r.status() == StatusCode::NOT_FOUND => {
+                    Ok(())
+                }
+                _ => Err(err),
+            })?;
```

Comment:
I found this a little quirky; figured I'd call it out.

- If we *haven't* run the later "volume creation" undo step, we need to cleanup this running snapshot.
- If we *have* run the later "volume creation" undo step, it's already gone